src/main/java/de/zalando/ep/zalenium/dashboard/Dashboard.java

```diff
@@ -250,29 +250,36 @@ public static void synchronizeExecutedTestsValues(File testCountFile) {
     }
 
     @VisibleForTesting
-    public static void loadTestInformationFromFile() {
+    public static List<TestInformation> loadTestInformationFromFile() {
         try {
-            if (executedTestsInformation.size() == 0) {
-                File testInformationFile = new File(getLocalVideosPath(), TEST_INFORMATION_FILE);
-                if (testInformationFile.exists()) {
-                    String testInformationContents = FileUtils.readFileToString(testInformationFile, UTF_8);
-                    Type collectionType = new TypeToken<ArrayList<TestInformation>>(){}.getType();
-                    executedTestsInformation = new Gson().fromJson(testInformationContents, collectionType);
-                }
+            List<TestInformation> testInformation = new ArrayList<>();
+            File testInformationFile = new File(getLocalVideosPath(), TEST_INFORMATION_FILE);
+            if (testInformationFile.exists()) {
+                String testInformationContents = FileUtils.readFileToString(testInformationFile, UTF_8);
+                Type collectionType = new TypeToken<ArrayList<TestInformation>>(){}.getType();
+                testInformation = new Gson().fromJson(testInformationContents, collectionType);
             }
+            return testInformation;
         } catch (Exception e) {
             LOGGER.warn(e.toString(), e);
         }
+        return new ArrayList<>();
     }
 
-    @VisibleForTesting
-    public static void dumpTestInformationToFile() {
+    public static void dumpTestInformationToFile(List<TestInformation> testInformationList) {
+        dumpTestInformationToFile(testInformationList, false);
+    }
+
+    public static void dumpTestInformationToFile(List<TestInformation> testInformationList, boolean replace) {
         try {
-            if (executedTestsInformation.size() > 0) {
-                File testInformationFile = new File(getLocalVideosPath(), TEST_INFORMATION_FILE);
-                Gson gson = new GsonBuilder().setPrettyPrinting().create();
-                FileUtils.writeStringToFile(testInformationFile, gson.toJson(executedTestsInformation), UTF_8);
+            File testInformationFile = new File(getLocalVideosPath(), TEST_INFORMATION_FILE);
+            List<TestInformation> testInformation = new ArrayList<>();
+            if (!replace) {
+                testInformation = loadTestInformationFromFile();
```

Comment:
So, it sounds like every 50 tests it will have to load in all of the existing tests to write the file? Doesn't that mean it will still use the same amount of memory? But it will just release it quicker?

How does that replace field work?  Maybe if you could configure it to only keep the last x tests, and then it just pops off the old ones?

I guess if there was a different data structure where you could just append to a file, and you didn't need to keep it in memory then that could make it easier.  Or maybe we've reached the point where we need a lightweight database or something?

Reply:
Yes, in the end it is just releasing the memory quicker so the garbage collector frees it sooner. When I wrote it I thought it was a good idea, but I think it just makes things more complicated. Afterwards I came to the same conclusion, it will be easier just to append to a file (I was just being lazy and wanted to keep the formatting through the Gson methods).

I will rework it and implement it in a more simple way.

Reply:
Thanks for the review btw.

Reply:
I reworked it using the append to a file approach, I am more satisfied with this one than the previous one.
This is the first step I wanted to take to actually render the dashboard from an API call instead of concatenating HTML files in Java. This should also allow people in the near future to use that API call to get the info they need to fetch videos or logs.